benchmarks/dali_vs_python/BM_rn50/model_repository/rn50_dali/pipeline.py

```diff
@@ -0,0 +1,57 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+# the Software, and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import nvidia.dali as dali
+import nvidia.dali.types as types
+
+
+def parse_args():
+    import argparse
+    parser = argparse.ArgumentParser(description="Serialize the pipeline and save it to a file")
+    parser.add_argument('file_path', type=str,
+                        help='The path where to save the serialized pipeline')
+    return parser.parse_args()
+
+
+def preprocessing(images):
+    images = dali.fn.decoders.image(images, device='mixed', output_type=types.RGB)
+    images = dali.fn.resize(images, resize_x=224, resize_y=224)
+    return dali.fn.crop_mirror_normalize(images,
+                                         dtype=types.FLOAT,
+                                         output_layout="HWC",
+                                         crop=(224, 224),
+                                         mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
+                                         std=[0.229 * 255, 0.224 * 255, 0.225 * 255])
+
+
+@dali.pipeline_def(batch_size=1, num_threads=1, device_id=0)
```

Comment:
Just a note: we might want to start setting the `batch_size` and `num_threads` values to the desired values here, because in the future we won't specify them in the model config

Reply:
But model config only sets the max batch size, right? Also, `batch_size` option in the decorator is in fact a `max_batch_size`. IIUC, the batch size should be actually inferred from the input, so I believe there's no need of setting the desired value here.

About the `num_threads`, in this case whatever is passed to the model config will overwrite the decorator setting. When the configuration is autogenerated, we will still need to default to some setting, and `1` seems to be a good default? Unless you have some other opinion on this?